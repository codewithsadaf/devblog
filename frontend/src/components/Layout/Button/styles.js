//imports
import styled, { css } from "styled-components";

const handleButtonVariant = {
    default: (theme) => css`
        background: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,
    outline: (theme) => css`
        background: transparent;
        color: ${theme.colors.primaryColor};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,

    success: (theme) => css`
        background: ${theme.colors.success};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.success};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,

    danger: (theme) => css`
        background: ${theme.colors.danger};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.danger};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,
};

const handleUppercase = () => css`
    text-transform: uppercase;
`;

const handleMaxWidth = () => css`
    width: 100%;
    padding: 1em 0;
`;

const handleSmallStyles = (theme) => css`
    padding: 7px 30px;
    border-radius: 4px;
    font-size: ${theme.font.sizes.small};
    border: 1px solid;

    &:hover {
        border: 1px solid;
    }
`;

export const ButtonContainer = styled.button`
    ${({ theme, buttonVariant, uppercase, maxWidth, isSmall }) => css`
        padding: 0.5em 1.8em;
        font-size: 1em;
        border: none;
        font-weight: bold;
        cursor: pointer;
        ${handleButtonVariant[buttonVariant](theme)};
        ${uppercase && handleUppercase()};
        ${maxWidth && handleMaxWidth()};
        ${isSmall && handleSmallStyles(theme)};
    `}
`;
