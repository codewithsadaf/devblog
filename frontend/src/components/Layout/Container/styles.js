//imports
import styled from "styled-components";

export const Container = styled.div`
    min-height: 60vh;
    margin-bottom: 5em;
    padding: 0 2em;
`;
