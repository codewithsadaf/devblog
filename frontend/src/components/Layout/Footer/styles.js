//imports
import styled, { css } from "styled-components";

export const FooterWrapper = styled.footer`
    ${({ theme }) => css`
        height: 250px;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background-color: ${theme.colors.footerBackground};
        text-align: center;
    `}
`;
export const FooterHeading = styled.h3`
    ${({ theme }) => css`
        @media ${theme.media.lteMedium} {
            font-size: ${theme.font.sizes.medium};
            line-height: ${theme.font.line_height.h6};
        }
    `}
`;

export const CopyrightText = styled.span`
    display: block;
    margin-top: 0.8em;
`;
