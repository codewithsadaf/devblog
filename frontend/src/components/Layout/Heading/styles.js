//imports
import styled, { css } from "styled-components";

const headingSize = {
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        line-height: ${theme.font.line_height.h1};
        ${mediaFont(theme)};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.huge};
        line-height: ${theme.font.line_height.h2};
    `,
    large: (theme) => css`
        font-size: ${theme.font.sizes.xxlarge};
        line-height: ${theme.font.line_height.h3};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.medium};
        line-height: ${theme.font.line_height.h6};
    `,
    small: (theme) => css`
        font-size: ${theme.font.sizes.small};
        line-height: ${theme.font.line_height.small};
    `,
    xsmall: (theme) => css`
        font-size: ${theme.font.sizes.xsmall};
        line-height: ${theme.font.line_height.small};
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    }
`;

export const Container = styled.h1`
    ${({ theme, size, uppercase, bold }) => css`
        text-transform: ${uppercase ? "uppercase" : "none"};
        font-weight: ${bold ? "bold" : 500};
        ${headingSize[size](theme)};
    `}
`;
