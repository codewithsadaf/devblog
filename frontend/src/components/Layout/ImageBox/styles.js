//imports
import styled, { css } from "styled-components";

const handleBigStyle = () => css`
    width: 100%;
    height: auto;
`;

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const ImageElm = styled.img`
    ${({ rounded }) => css`
        width: 150px;
        height: 150px;
        object-fit: cover;
        display: block;
        margin: 1em auto;
        border-radius: ${rounded ? "50%" : "7px"};
    `}
`;
