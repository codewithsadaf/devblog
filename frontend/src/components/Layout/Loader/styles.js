//imports
import styled, { css } from "styled-components";

export const Container = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100vh;
    background: rgba(0, 0, 0, 0.6);
    z-index: 99999;
    pointer-events: all;
`;

export const Circle = styled.div`
    ${({ theme }) => css`
        &,
        &::before,
        &::after {
            border-radius: 50%;
            width: 2.5em;
            height: 2.5em;
            animation-fill-mode: both;
            animation: bblFadInOut 1.8s infinite ease-in-out;
        }

        & {
            color: ${theme.colors.white};
            font-size: 7px;
            position: relative;
            text-indent: -9999em;
            transform: translateZ(0);
            animation-delay: -0.16s;
        }

        &::before,
        &:after {
            content: "";
            position: absolute;
            top: 0;
        }

        &::before {
            left: -3.5em;
            animation-delay: -0.32s;
        }

        &::after {
            left: 3.5em;
        }

        @keyframes bblFadInOut {
            0%,
            80%,
            100% {
                box-shadow: 0 2.5em 0 -1.3em;
            }
            40% {
                box-shadow: 0 2.5em 0 0;
            }
        }
    `}
`;
