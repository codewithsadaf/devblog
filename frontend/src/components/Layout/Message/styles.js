//imports
import styled, { css } from "styled-components";
//react icons
import { IoClose } from "react-icons/io5";

export const Container = styled.div`
    ${({ theme }) => css`
        display: flex;
        align-items: center;
        justify-content: space-between;
        position: fixed;
        bottom: 30px;
        left: 50%;
        transform: translateX(-50%);
        padding: 25px 30px;
        min-width: 25%;
        border-radius: 7px;
        z-index: 9999;
        font-weight: bold;
        box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
        transition: ${theme.transition.tran_03};

        &.success {
            background-color: ${theme.colors.success};
            color: ${theme.colors.white};
        }

        &.error {
            background-color: ${theme.colors.danger};
            color: ${theme.colors.white};
        }

        @media ${theme.media.lteMedium} {
            bottom: 0;
            right: 50%;
            left: 50%;
            min-width: 90%;
            transform: translate(-50%, -50%);
        }
    `}
`;

export const CloseIcon = styled(IoClose)`
    ${({ theme }) => css`
        color: ${theme.colors.white};
        cursor: pointer;
        font-size: 19px;
        margin-left: 0.3em;
    `}
`;

export const MessageText = styled.span`
    display: block;
`;
