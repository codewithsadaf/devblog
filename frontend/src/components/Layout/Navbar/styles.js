//imports
import { Link } from "react-router-dom";
import styled, { css } from "styled-components";
//react icons
import { MdClose } from "react-icons/md";
import { GiHamburgerMenu } from "react-icons/gi";

export const Nav = styled.nav`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    height: 70px;
    padding: 1.2em 2em;
    box-shadow: 0 2px 10px 0 rgba(0, 0, 0, 0.15);
    position: relative;
`;

export const Nav_brand = styled(Link)`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.large};
        color: ${theme.colors.primaryColor};
    `}
`;

export const bold_Effect = styled.span`
    font-weight: 900;
    text-transform: uppercase;
`;

export const Menu = styled.ul`
    ${({ theme, open }) => css`
        display: flex;

        @media ${theme.media.lteMedium} {
            position: fixed;
            top: 0;
            right: ${open ? "0" : "-300px"};
            height: 100vh;
            width: 300px;
            display: flex;
            flex-direction: column;
            background: ${theme.colors.primaryColor};
            padding: 30px;
            transition: ${theme.transition.tran_05};
            z-index: 999999;
        }
    `}
`;

export const List_Item = styled.li`
    ${({ theme }) => css`
        margin-left: ${theme.spacings.medium};

        @media ${theme.media.lteMedium} {
            margin-left: 0;
            margin-top: ${theme.spacings.medium};
        }
    `}
`;

export const HamburgerIcon = styled(GiHamburgerMenu)`
    ${({ theme }) => css`
        font-size: 26px;
        display: none;
        visibility: hidden;
        opacity: 0;
        pointer-events: none;
        cursor: pointer;

        @media ${theme.media.lteMedium} {
            display: block;
            visibility: visible;
            opacity: 1;
            pointer-events: all;
        }
    `}
`;

export const CloseIcon = styled(MdClose)`
    ${({ theme }) => css`
        font-size: 26px;
        display: none;
        visibility: hidden;
        opacity: 0;
        pointer-events: none;
        cursor: pointer;
        color: ${theme.colors.white};

        @media ${theme.media.lteMedium} {
            display: block;
            visibility: visible;
            opacity: 1;
            pointer-events: all;
            border-radius: 50%;
        }
    `}
`;

export const LogoutText = styled.span`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.default};
        padding: 0.4em 0.5em;
        color: ${theme.colors.primaryColor};
        font-weight: 500;
        transition: ${theme.transition.tran_03};
        text-transform: capitalize;
        cursor: pointer;

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }

        @media ${theme.media.lteMedium} {
            color: ${theme.colors.white};
            font-size: ${theme.font.sizes.medium};
        }
    `}
`;
