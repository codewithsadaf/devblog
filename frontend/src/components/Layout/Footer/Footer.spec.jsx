//imports
import { screen } from "@testing-library/react";
//styles
import { renderTheme } from "../../../styles/render-theme";
//components
import { Footer } from "./";

describe("<Footer />", () => {
    describe("Rendering", () => {
        it("should render Footer", () => {
            renderTheme(<Footer />);

            const heading = screen.getByRole("heading", {
                name: "Escreva sobre o que você tem interesse!",
            });

            expect(heading).toBeInTheDocument();
        });

        it("should match snapshot", () => {
            const { container } = renderTheme(<Footer />);

            expect(container.firstChild).toMatchSnapshot();
        });
    });

    describe("Functionality", () => {
        it("should render Footer copyright text with current year", () => {
            renderTheme(<Footer />);
            const currentYear = new Date().getFullYear();

            const currentYearText = screen.getByText(
                `DevBlog - Gustavo Matos © ${currentYear}`
            );

            expect(currentYearText).toBeInTheDocument();
        });
    });
});
