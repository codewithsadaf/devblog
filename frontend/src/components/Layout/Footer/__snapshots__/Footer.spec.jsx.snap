// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Rendering should match snapshot 1`] = `
.c0 {
  height: 250px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #edf3f6;
  text-align: center;
}

.c2 {
  display: block;
  margin-top: 0.8em;
}

@media (max-width:768px) {
  .c1 {
    font-size: 1.2em;
    line-height: 120%;
  }
}

<footer
  class="c0"
>
  <h3
    class="c1"
  >
    Escreva sobre o que você tem interesse!
  </h3>
  <span
    class="c2"
  >
    DevBlog - Gustavo Matos © 
    2023
  </span>
</footer>
`;
